### PR TITLE
Refactor: Update component previews

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
The purpose of this PR is to clean up the way previews are generated for our view components.  Currently previews are generated using rails markup which leads to an incorrect representation of how to use the component in the LookBook source panel.  This update changes to using erb files to generate the preview, which is the same syntax that is used to write the component in regular view files allowing for easier adoption of the components.

* If the component is prefixed with `viral_` it is part of the _Viral Component Library_

## Screenshots or screen recordings

**Source Before:**
image

**Source After:**
image

## How to set up and validate locally

All updated components previews can be viewed within our [LookBook](http://localhost:3000/rails/lookbook/) while the development server is running.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the PR acceptance checklist for this PR.

### CODE REVIEW

File: app/components/viral/dropdown_component.rb [hunk 1]
```diff
@@ -20,6 +20,7 @@ def initialize(label: nil, icon: nil, caret: false, trigger: TRIGGER_DEFAULT,
       @trigger = TRIGGER_MAPPINGS[trigger]
 
       @system_arguments = system_arguments
+      @system_arguments[:id] ||= "dd-#{SecureRandom.hex(4)}"
```
Comment (deepsidhu85) [inline app/components/viral/dropdown_component.rb:23]: Could this possibly lead to an issue of duplicate ids?

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks good to me just the one question below. Great work on the component library so far!

File: app/components/viral/dropdown_component.rb (hunk 1 above)
Comment (joshsadam) [reply app/components/viral/dropdown_component.rb]: I increased it to 10 so I think that there should any duplicates. Thanks

Comment (joshsadam) [reply app/components/viral/dropdown_component.rb]: I increased it to 10 so I think that there should any duplicates. Thanks

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**